brave/brave-core
# Update Geolocation permission bubble

fix https://github.com/brave/brave-browser/issues/16897

Added more information to users for getting precise location.
With passed `enableHighAccuracy`, permission bubble could give more precise info.

When page requests current position **_w/o_** `highAccuracy`, below bubble should shown.
[**Type A**]
![image](https://github.com/brave/brave-core/assets/6786187/08e3bf21-6309-4bda-964e-584630e5b841)

When page requests current position **_with_** `highAccuracy`, we have two types of bubble.
When os location service is **_enabled_**, below bubble is shown
[**Type B**]
![image](https://github.com/brave/brave-core/assets/6786187/aff26bf9-de97-4b8e-b443-6cffd70d2489)
When os location service is **_disabled_**, this is shown.
[**Type C**]
![image](https://github.com/brave/brave-core/assets/6786187/a2aaa0e4-47c6-4326-abf8-13cbbbe4e25c)



Resolves



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

`GeolocationPermissionRequestBrowserTest.SetEnabledHighAccuracyTest`
`LocalizationUtilTest.GetLocalizedResourceUTF16StringWithPlaceholders`

_NOTE_: geolocation permission bubble is not changed on Linux

1. Launch browser and open inspector with any tabs
2. Run `navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : false });`
3. check **Type A** bubble is shown
4. Dismiss bubble and reset permission state if needed
5. Disable OS location service
6. Run `navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : true });`
7. Chek **Type B** bubble is shown
8. Dismiss bubble and reset permission state if needed
9. Enable OS location service (On macOS, brave should be enabled also)
10. Run `navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : true });`
11. Check **Type C** bubble is shown
